File: src/services/authService.js
import api from './api';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { TOKEN_KEY } from '../constants/api';

export const login = async (email, password) => {
  const response = await api.post('/auth/login', { email, password });
  if (response.data.token) {
    await AsyncStorage.setItem(TOKEN_KEY, response.data.token);
  }
  return response.data;
};

export const register = async (userData) => {
  const response = await api.post('/auth/register', {
    ...userData,
    role: 'consumer',
  });
  if (response.data.token) {
    await AsyncStorage.setItem(TOKEN_KEY, response.data.token);
  }
  return response.data;
};

export const googleLogin = async (token) => {
  const response = await api.post('/auth/google', { token });
  if (response.data.token) {
    await AsyncStorage.setItem(TOKEN_KEY, response.data.token);
  }
  return response.data;
};

export const getCurrentUser = async () => {
  const response = await api.get('/auth/me');
  return response.data;
};

export const logout = async () => {
  await AsyncStorage.removeItem(TOKEN_KEY);
};

export const getStoredToken = async () => {
  return await AsyncStorage.getItem(TOKEN_KEY);
};

export const updateProfile = async (profileData) => {
  const response = await api.put('/users/profile', profileData);
  return response.data;
};

export const uploadAvatar = async (imageUri) => {
  const formData = new FormData();
  formData.append('avatar', {
    uri: imageUri,
    type: 'image/jpeg',
    name: 'avatar.jpg',
  });
  
  const response = await api.post('/users/avatar', formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  });
  return response.data;
};

File: lib/services/auth_service.dart
import 'api_service.dart';
import '../constants/api.dart';

class AuthService {
  static Future<Map<String, dynamic>> login(
      String email, String password) async {
    final data = await ApiService.post(
      ApiConstants.login,
      {'email': email, 'password': password},
      auth: false,
    );
    await ApiService.saveToken(data['token'] ?? data['accessToken']);
    return data;
  }

  static Future<Map<String, dynamic>> register(
      Map<String, dynamic> userData) async {
    userData['role'] = 'consumer';
    final data = await ApiService.post(
      ApiConstants.register,
      userData,
      auth: false,
    );
    await ApiService.saveToken(data['token'] ?? data['accessToken']);
    return data;
  }

  static Future<Map<String, dynamic>> getMe() async {
    return await ApiService.get(ApiConstants.me);
  }

  static Future<void> logout() async {
    try {
      await ApiService.post(ApiConstants.logout, {});
    } catch (_) {}
    await ApiService.clearToken();
  }

  static Future<bool> isLoggedIn() async {
    final token = await ApiService.getToken();
    return token != null;
  }
}
